controllers/storagecluster/cephcluster.go

```diff
@@ -1483,6 +1483,26 @@ func isEncrptionSettingUpdated(clusterWideEncrytion bool, existingDeviceSet []ro
 	return false
 }
 
+// setDefaultMetadataPoolSpec sets the common pool spec for all metadata pools as necessary
+func setDefaultMetadataPoolSpec(poolSpec *rookCephv1.PoolSpec, sc *ocsv1.StorageCluster) {
```

Comment:
I only see this method being called for the `.nfs` and `.mgr` pools, which don't have any spec exposed in the storageCluster CR, right? If that's the intended case (and not for the filesystem and object store), seems like this method can be simplified. For example, device class and failure domain would always be `""`, so no need for those checks on lines 1489 and 1492.

Reply:
We will use setDefaultMetadataPoolSpec() for all the metadata pools